src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -128,27 +142,92 @@ private bool TryGetFunctionNameFromActivityInvocation(InvocationExpressionSyntax
             return false;
         }
 
-        private bool TryGetInputNodeFromCallActivityInvocation(InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        private static bool TryGetInputNodeFromCallActivityInvocation(SemanticModel semanticModel, InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        {
+            // If CallActivity is a custom method in user code
+            if (SyntaxNodeUtils.TryGetDeclaredSyntaxNode(semanticModel, invocationExpression, out SyntaxNode declaration))
+            {
+                if (TryGetSpecificParameterIndex(declaration, "object input", out int inputParameterIndex))
```

Comment:
Whoa, great point. I did this before I refactored OrchestratorAnalyzer/knew I could pull it off with FunctionAnalyzer. I didn't realize that it also covers this case.

Because of this, I just have to change this to return/back out of analysis if the CallActivity method is defined in customer code (on this TryGetDeclaredSyntaxNode method). Alternatively, I could check if the methods parameters match one of our Durable methods (and if it doesnt match, back out).

Reply:
(deleted my last comment) I do need this. The Durable CallActivity method will always be correct if a customer is using an extension method. I want to help customers who are passing in incorrectly typed function names or using the wrong input, so I need to identify which argument represents their input. Otherwise, this analyzer wont work for them at all. I know that I can't cover all cases where a customer is using an extension method, but I'm doing a "best effort" type check here that will get the input off of the method declaration if its "object input." Otherwise, it could be anything and I don't continue analysis on specifically the input analyzer (the function name analyzer will still work in this case).

Reply:
I didn't understand this statement:
> The Durable CallActivity method will always be correct if a customer is using an extension method.

Why would it always be correct? People can still write bad code inside of their extension methods.

It's likely I'm not understanding the issue correctly. I don't want to hold up this PR on this since this was more of a question and I don't think it's really an issue of correctness. I'm fine with keeping it as-is as long as you feel confident about it.

Reply:
You're right, that's a weird sentence. I think it's sort of difficult to explain what I meant without giving a more thorough explanation of what I'm actually analyzing here.

What I meant to say is that there will (likely) be nothing to flag in an extension method.

Take this example (how I assume customers will probably use extension methods):
`public static Task<TResult> CallActivityAsync<TResult>(
    this IDurableOrchestrationContext context, string functionName, object input, RetryOptions retryOptions
)
        {
            return retryOptions != null
                ? context.CallActivityWithRetryAsync<TResult>(functionName, retryOptions, input)
                : context.CallActivityAsync<TResult>(functionName, input);
        }`

functionName isn't the name of a function. I cannot determine which function the user is trying to call here. I assume that most extension methods will work this way. The parameters will be used as arguments for the Durable methods. So there is no code to flag in this method. I actually don't even store this as an ActivityFunctionCall because it doesn't have a functionName.

We want to help customers wherever they are calling specific functions to make sure the name matches a real activity function (and the input or return type match that function). So we need to analyze the place in code where the arguments are actually being passed in. And by actually I mean where I can determine what the values are.

As in this part of the code:
`await context.CallActivityAsync<string>("E1_SayHello_DirectInput", "London", somethingCustom)`

If a customer is using an extension method like this, we will still analyze the extension method's Durable version of CallActivity* to make sure the arguments match an existing activity function:
`public static Task<TResult> CallActivityAsync<TResult>(
    this IDurableOrchestrationContext context)
        {
            return context.CallActivityAsync<TResult>("E1_SayHello", "World");
        }`

On the topic of extension methods...
There is technically one case where I will possibly incorrectly flag code in an extension method.

`public static Task<TResult> CallActivityAsync<TResult>(
    this IDurableOrchestrationContext context, object input)
        {
            return context.CallActivityAsync<TResult>("E1_SayHello", input);
        }`
        
I don't expect this particular problem to come up...
(input and TResult would be flagged here, to fix this input would either need to be cast as a string or the parameter changed from object to string and TResult would also have to be changed to string)